### DISCUSSION TITLE
Nuxt 3 Dynamic Routing Conflict with Nested and Single Slug Pages

### DISCUSSION THREAD

**Discussion by FrazeColder** [nuxt/nuxt#27420] [Questions]
I am building a project with Nuxt 3, However, I am facing a routing conflict with my current setup. Here is my folder structure:

```
pages/
  [types]/
    [slug].vue
  [...categoryPath]/
    [slug].vue
  [slug].vue
```

Requirements:

- Product Details Pages: `pages/[types]/[slug].vue` should handle URLs like `/deal/:slug`, `/voucher/:slug`, and `/microjob/:slug`. Therefore, I would like to restrict `[type]` to deal, voucher and microjob only. 

- Blog Posts with Nested Categories: `pages/[...categoryPath]/[slug].vue` should handle any other nested category paths like `/finance/:slug`, `/finance/markets/:slug`, `/finance/markets/europe/:slug` and so on except for the specified types (deal, voucher and microjob).

- Single Slug Pages: `pages/[slug].vue` should handle single slug pages like `/imprint`, `/privacy-policy`, or any other single slug. But that is already the case and no problem at the moment. 

Issue:

All routes which have and path like `deal/my-slug-12`, `voucher/my-slug` or also `market/my-slug` as well as `finance/my-slug` are being handled by `pages/[types]/[slug].vue` which is wrong. Only paths like starting with `deal/...`, `voucher/...` and `microjob/...` should be handled by `pages/[types]/[slug].vue`. 

If the path is e.g. `market/finance/my-slug` or has a longer category path like `market/finance/europe/germany/my-slug` the route is being handled correctly by `pages/[...categoryPath]/[slug].vue`. But one segment routes are always being handled by `pages/[types]/[slug].vue` which is wrong as they should only handle routes starting with deal, voucher and microjob.

How can I archive this behavior?

**Comment by CedsTrash** [comment]
@FrazeColder Did you eventually find a solution to this problem? I'm facing the same issue. Thanks.
